### PR TITLE
Bump jsonschema version to 4.18.0

### PR DESCRIPTION
**Pull request by hussein-awala**
related: #32434

Bump the min jsonschema version to 4.18.0 to improve its performance ([more details](https://github.com/python-jsonschema/jsonschema/issues/941#issuecomment-1441416922))

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Review by tirkarthi — APPROVED** [review]

**Comment by hussein-awala** [comment]
https://github.com/python-openapi/openapi-schema-validator/commit/5bf8b10416cded4c2b7fc3b19d854c36036011fb

I will try to bump `openapi-schema-validator` version in amazon provider

**Comment by hussein-awala** [comment]
and there is a core test which fail with python 3.11:
```
FAILED tests/models/test_param.py::TestParam::test_string_date_format[20120503] - airflow.exceptions.ParamValidationError: '20120503' is not a 'date'

Failed validating 'format' in schema:
    {'format': 'date', 'type': 'string'}

On instance:
    '20120503'
```

**Comment by tirkarthi** [comment]
https://github.com/python-jsonschema/jsonschema/issues/1056

File: tests/providers/amazon/aws/hooks/test_cloud_formation.py [hunk 1]
```diff
@@ -18,11 +18,89 @@
 from __future__ import annotations
 
 import json
-
-from moto import mock_cloudformation
+from unittest import mock
 
 from airflow.providers.amazon.aws.hooks.cloud_formation import CloudFormationHook
 
+# TODO: restore this test with moto when openapi_schema_validator 0.6.0 is released
```
Comment (tirkarthi) [inline tests/providers/amazon/aws/hooks/test_cloud_formation.py:25]: Is using pytest.mark.skip a better option here?

**Comment by potiuk** [comment]
needs tests fixing :(

**Pull request merged**